tests/helpers/load_state_tests.py

```diff
@@ -79,10 +246,28 @@ def run_blockchain_st_test(test_case: Dict, load: Load) -> None:
                 break
 
         if block_exception:
-            # TODO: Once all the specific exception types are thrown,
-            #       only `pytest.raises` the correct exception type instead of
-            #       all of them.
-            with pytest.raises((EthereumException, RLPException)):
+            # TODO: Once all the specific exception types for Invalid Block
+            #       are thrown, only `pytest.raises` the correct exception
+            # type instead of all of them.
+            fork_name = load.fork.fork_module
+            common_exceptions = (
+                InvalidBlock,
+                InsufficientTransactionGasError,
+                NonceTooHighError,
+                GasUsedExceedsLimitError,
+                NonceMismatchError,
+                InsufficientBalanceError,
+                InvalidSenderError,
+                InvalidSignatureError,
+                RLPException,
+            )
+            fork_specific_exceptions = FORK_SPECIFIC_EXCEPTIONS.get(
+                fork_name,
+                (),
+            )
+            all_exceptions = common_exceptions + fork_specific_exceptions
+
+            with pytest.raises(all_exceptions):
```

Comment:
I believe the JSON fixture contains a string representing the expected exception. Is it possible to map that to exactly one exception type?

Reply:
I found that the block_exceptions from legacy tests/blockchain test fixtures are very different from the state transition exceptions. (and the state_transition method from the fork is used to raise the error caught here).
 
This specific function also seems to work with blockchaintests fixtures so couldn't account for the expected exception from state_tests. If i'm thinking wrong about this - do lmk!